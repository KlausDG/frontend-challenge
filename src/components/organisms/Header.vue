<script setup lang="ts">
import type { Header } from '@/models'
import { Stack, Typography } from '@/components/atoms'
import { IconText } from '@/components/molecules'

defineProps<{
  data: Header
}>()
</script>

<template>
  <section class="header__container">
    <div class="header__left_section">
      <Typography tag="h2" color="secondary">Pre-Order</Typography>
      <Typography tag="h4" color="secondary">{{ data.number }}</Typography>
      <Typography color="secondary">#ME{{ data.serial }}</Typography>
    </div>
    <Stack direction="column" :spacing="12">
      <Typography tag="h5" color="dark" class="title">{{ data.buyer }}</Typography>
      <Stack direction="column" :spacing="6">
        <IconText icon="user" :text="data.contact.name" />
        <Stack :spacing="16">
          <IconText icon="envelope" :text="data.contact.email" />
          <IconText icon="phone" :text="data.contact.phone" />
          <IconText icon="print" :text="data.contact.fax" />
        </Stack>
      </Stack>
    </Stack>
    <Stack direction="column" :spacing="6" class="header__right_section">
      <Typography tag="h5" color="dark">{{ `${data.currency} ${data.price}` }}</Typography>
      <Typography tag="h5" color="success">{{ data.status }}</Typography>
      <Stack :spacing="4" class="text_icon">
        <Typography color="dark">Crated at {{ data.createdAt }}</Typography>
        <unicon name="exclamation-circle" icon-style="solid" fill="#A3ACB6" />
      </Stack>
    </Stack>
  </section>
</template>

<style lang="scss" scoped>
.header__container {
  display: flex;
  gap: 32px;

  & > div {
    flex: 1;
  }
}

.header__left_section {
  background-color: $color-primary;
  max-width: 200px;
  border-radius: 0 8px 8px 0;
  padding: 16px;
}

.header__right_section {
  text-align: end;
}

.text_icon {
  align-items: center;
  justify-content: end;
}
@media (min-width: 1024px) {
}
</style>
