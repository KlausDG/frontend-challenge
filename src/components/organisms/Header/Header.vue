<script setup lang="ts">
import type { Header } from '@/models'
import { Stack, Typography } from '@/components/atoms'
import { IconText } from '@/components/molecules'

defineProps<{
  data: Header
}>()
</script>

<template>
  <section class="header__container">
    <div class="header__left_section">
      <Typography tag="h2" color="secondary">Pre-Order</Typography>
      <Typography tag="h4" color="secondary">{{ data.number }}</Typography>
      <Typography color="secondary">#ME{{ data.serial }}</Typography>
    </div>
    <Stack direction="column" :spacing="12" class="header__middle_section">
      <Typography tag="h5" color="dark" class="title">{{ data.buyer }}</Typography>
      <Stack direction="column" :spacing="6">
        <IconText icon="user" :text="data.contact.name" />
        <Stack :spacing="16">
          <IconText icon="envelope" :text="data.contact.email" />
          <IconText icon="phone" :text="data.contact.phone" />
          <IconText icon="print" :text="data.contact.fax" />
        </Stack>
      </Stack>
    </Stack>
    <Stack direction="column" :spacing="6" class="header__right_section">
      <Typography tag="h5" color="dark">{{ `${data.currency} ${data.price}` }}</Typography>
      <Typography tag="h5" color="success">{{ data.status }}</Typography>
      <Stack :spacing="4" class="text_icon">
        <Typography color="dark">Crated at {{ data.createdAt }}</Typography>
        <unicon name="exclamation-circle" icon-style="solid" fill="#A3ACB6" />
      </Stack>
    </Stack>
  </section>
</template>

<style lang="scss" scoped>
.header__container {
  display: flex;
  gap: $spacing-xl;

  & > div {
    flex: 1;
  }

  @media (max-width: 1024px) {
    flex-direction: column;
    gap: $spacing-sm;
  }
}

.header__left_section {
  background-color: $color-primary;
  max-width: 200px;
  border-radius: 0 $radius-md $radius-md 0;
  padding: $spacing-md;
  order: 1;

  @media (max-width: 1024px) {
    max-width: none;
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-radius: 0 0 $radius-md $radius-md;
  }
}

.header__middle_section {
  order: 2;

  @media (max-width: 1024px) {
    order: 3;
    padding-left: $spacing-md;
    padding-right: $spacing-md;
  }
}

.header__right_section {
  text-align: end;
  order: 3;

  @media (max-width: 1024px) {
    text-align: start;
    order: 1;
    padding-left: $spacing-md;
    padding-right: $spacing-md;
  }
}

.text_icon {
  align-items: center;
  justify-content: end;

  @media (max-width: 1024px) {
    justify-content: start;
  }
}

@media (min-width: 1024px) {
}
</style>
