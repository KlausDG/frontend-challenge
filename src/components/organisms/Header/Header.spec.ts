import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import Header from './Header.vue'

const TypographyMock = {
  name: 'Typography',
  props: ['tag', 'color'],
  template: '<div class="mock-typography"><slot /></div>',
}

const StackMock = {
  name: 'Stack',
  props: ['direction', 'spacing'],
  template: '<div class="mock-stack"><slot /></div>',
}

const IconTextMock = {
  name: 'IconText',
  props: ['icon', 'text'],
  template: '<div class="mock-icon-text">{{ icon }}:{{ text }}</div>',
}

const UniconMock = {
  name: 'unicon',
  props: ['name', 'iconStyle', 'fill'],
  template: '<div class="mock-unicon">{{ name }}</div>',
}

describe('Header.vue', () => {
  const headerData = {
    number: '123',
    serial: '456',
    buyer: 'John Doe',
    contact: {
      name: 'Jane Doe',
      email: 'jane@example.com',
      phone: '123456789',
      fax: '987654321',
    },
    currency: 'USD',
    price: 99.99,
    status: 'Paid',
    createdAt: '2025-08-27',
  }

  const wrapper = mount(Header, {
    props: { data: headerData },
    global: {
      components: {
        Typography: TypographyMock,
        Stack: StackMock,
        IconText: IconTextMock,
        unicon: UniconMock,
      },
    },
  })

  it('renders left section with Pre-Order and data', () => {
    const leftSection = wrapper.find('.header__left_section')
    expect(leftSection.exists()).toBe(true)
    expect(leftSection.text()).toContain('Pre-Order')
    expect(leftSection.text()).toContain(headerData.number)
    expect(leftSection.text()).toContain(`#ME${headerData.serial}`)
  })

  it('renders middle section with buyer and contact info', () => {
    const middleSection = wrapper.find('.header__middle_section')
    expect(middleSection.exists()).toBe(true)
    expect(middleSection.text()).toContain(headerData.buyer)

    const iconTexts = middleSection.findAllComponents(IconTextMock)
    expect(iconTexts).toHaveLength(4)
    expect(iconTexts[0].props('icon')).toBe('user')
    expect(iconTexts[0].props('text')).toBe(headerData.contact.name)
  })

  it('renders right section with price, status, and createdAt', () => {
    const rightSection = wrapper.find('.header__right_section')
    expect(rightSection.exists()).toBe(true)
    expect(rightSection.text()).toContain(`${headerData.currency} ${headerData.price}`)
    expect(rightSection.text()).toContain(headerData.status)
    expect(rightSection.text()).toContain(headerData.createdAt)

    const unicon = rightSection.findComponent(UniconMock)
    expect(unicon.exists()).toBe(true)
    expect(unicon.props('name')).toBe('exclamation-circle')
  })

  it('matches the component snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
