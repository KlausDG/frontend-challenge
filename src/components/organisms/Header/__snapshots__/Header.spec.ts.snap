// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header.vue > matches the component snapshot 1`] = `
"<section data-v-ae2a1f96="" class="header__container">
  <div data-v-ae2a1f96="" class="header__left_section">
    <h2 data-v-ae2a1f96="" class="typography _h2_d0cbd2 _secondary_d0cbd2">Pre-Order</h2>
    <h4 data-v-ae2a1f96="" class="typography _h4_d0cbd2 _secondary_d0cbd2">123</h4>
    <p data-v-ae2a1f96="" class="typography _caption_d0cbd2 _secondary_d0cbd2">#ME456</p>
  </div>
  <div data-v-69f42c1a="" data-v-ae2a1f96="" class="stack__container header__middle_section" style="flex-direction: column; gap: 12px;">
    <h5 data-v-ae2a1f96="" class="typography _h5_d0cbd2 _dark_d0cbd2 title">John Doe</h5>
    <div data-v-69f42c1a="" data-v-ae2a1f96="" class="stack__container" style="flex-direction: column; gap: 6px;">
      <div data-v-70c01fa4="" data-v-ae2a1f96="" class="icon-text__container">
        <div data-v-70c01fa4="" class="mock-unicon">user</div>
        <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">Jane Doe</p>
      </div>
      <div data-v-69f42c1a="" data-v-ae2a1f96="" class="stack__container" style="flex-direction: row; gap: 16px;">
        <div data-v-70c01fa4="" data-v-ae2a1f96="" class="icon-text__container">
          <div data-v-70c01fa4="" class="mock-unicon">envelope</div>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">jane@example.com</p>
        </div>
        <div data-v-70c01fa4="" data-v-ae2a1f96="" class="icon-text__container">
          <div data-v-70c01fa4="" class="mock-unicon">phone</div>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">123456789</p>
        </div>
        <div data-v-70c01fa4="" data-v-ae2a1f96="" class="icon-text__container">
          <div data-v-70c01fa4="" class="mock-unicon">print</div>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">987654321</p>
        </div>
      </div>
    </div>
  </div>
  <div data-v-69f42c1a="" data-v-ae2a1f96="" class="stack__container header__right_section" style="flex-direction: column; gap: 6px;">
    <h5 data-v-ae2a1f96="" class="typography _h5_d0cbd2 _dark_d0cbd2">USD 99.99</h5>
    <h5 data-v-ae2a1f96="" class="typography _h5_d0cbd2 _success_d0cbd2">Paid</h5>
    <div data-v-69f42c1a="" data-v-ae2a1f96="" class="stack__container text_icon" style="flex-direction: row; gap: 4px;">
      <p data-v-ae2a1f96="" class="typography _caption_d0cbd2 _dark_d0cbd2">Crated at 2025-08-27</p>
      <div data-v-ae2a1f96="" class="mock-unicon">exclamation-circle</div>
    </div>
  </div>
</section>"
`;
